backend/coreapp/views/scratch.py

```diff
@@ -427,6 +435,58 @@ def export(self, request: Request, pk):
             },
         )
 
+    @action(detail=True, methods=["POST"])
+    def pr(self, request: Request, pk):
+        scratch: Scratch = self.get_object()
+        if scratch.owner != request.profile:
+            raise ScratchNotOwnerException()
+        # TODO: catch exceptions
+        fn: ProjectFunction = scratch.project_function
+        token = request.profile.user.github.access_token
+        github_repo: Repository = fn.project.repo.details(token)
+
+        # Get or create fork
+        try:
+            # TODO: likely need to pull this to make it up to date
+            fork = Github(token).get_repo(f"{scratch.owner}/{github_repo.name}")
+        except UnknownObjectException:
+            # TODO: maybe just call this? POST == GET if exists?
+            fork = Github(token).get_user().create_fork(github_repo)
+
+        # Change file contents
+        contents = fork.get_contents(fn.src_file)
+        new_content = re.compile(
+            # TODO: escape function name?
+            rf"INCLUDE_ASM\([^,]+, [^,]+, {fn.display_name}[^\)]*\);",
+            re.MULTILINE,
+        ).sub(
+            scratch.source_code,
+            contents.content,
+        )
+
+        # Make commit
+        # TODO: randomize and sanitize?
+        fork_branch = f"decompme__GEN__{fn.display_name}"
```

Comment:
So a branch is required?

Reply:
Not strictly, but it would guarantee no merge conflicts. (Another solution would be force-pushing, but that's obviously dangerous.)